server/eclipse-project/src/test/java/au/org/democracydevelopers/corla/endpoint/GetAssertionsTests.java

```diff
@@ -0,0 +1,248 @@
+
+    private final List<Choice> boulderMayoralCandidates = List.of(
+            new Choice("Aaron Brockett", "", false, false),
+            new Choice("Nicole Speer", "", false, false),
+            new Choice("Bob Yates", "", false, false),
+            new Choice("Paul Tweedlie", "", false, false)
+    );
+
+    /**
+     * Two IRV Contests for mocking the IRVContestCollector.
+     */
+    private final Contest tinyIRVExample = new Contest(tinyIRV, new County("Arapahoe", 3L), ContestType.IRV.toString(),
+            List.of(alice, bob, chuan), 3, 1, 0);
+    private final ContestResult tinyIRVContestResult = new ContestResult(tinyIRV);
+    private final Contest boulderMayoralContest = new Contest(boulderMayoral, new County("Boulder", 7L), ContestType.IRV.toString(),
+            boulderMayoralCandidates, 4, 1, 0);
+    private final ContestResult boulderIRVContestResult = new ContestResult(boulderMayoral);
+    private final List<ContestResult> mockedIRVContestResults = List.of(boulderIRVContestResult, tinyIRVContestResult);
+
+
+    /**
+     * Wiremock server for mocking the raire service.
+     */
+    final WireMockServer wireMockRaireServer = new WireMockServer();
+
+    /**
+     * Initialise mocked objects prior to the first test.
+     */
+    @BeforeClass
+    public void initMocks() {
+        MockitoAnnotations.openMocks(this);
+
+        boulderIRVContestResult.setAuditReason(AuditReason.COUNTY_WIDE_CONTEST);
+        boulderIRVContestResult.setBallotCount(100000L);
+        boulderIRVContestResult.setWinners(Set.of("Aaron Brockett"));
+        boulderIRVContestResult.addContests(Set.of(boulderMayoralContest));
+
+        tinyIRVContestResult.setAuditReason(AuditReason.COUNTY_WIDE_CONTEST);
+        tinyIRVContestResult.setBallotCount(10L);
+        tinyIRVContestResult.setWinners(Set.of("Alice"));
+        tinyIRVContestResult.addContests(Set.of(tinyIRVExample));
+
+        wireMockRaireServer.start();
+        configureFor("localhost", wireMockRaireServer.port());
+        stubFor(post(urlEqualTo("/raire/get-assertions-csv"))
+                .willReturn(aResponse()
+                        .withStatus(200)
+                        .withHeader("Content-Type", "application/octet-stream")
+                        .withBody("Test csv")));
+        stubFor(post(urlEqualTo("/raire/get-assertions-json"))
+                .willReturn(aResponse()
+                .withStatus(200)
+                .withHeader("Content-Type", "application/json")
+                .withBody("Test json")));
+    }
+
+    @AfterClass
+    public void closeMocks() {
+        wireMockRaireServer.stop();
+    }
+
+    /**
+     * Calls the getAssertions main endpoint function, for csv, and checks that the right file names are present in the
+     * zip.
+     * @throws Exception never.
+     */
+    @Test
+    public void rightFileNamesInZipCSV() throws Exception {
+        testUtils.log(LOGGER, "rightFileNamesInZipCSV");
+
```

Comment:
The following try block can be abstracted to a function that is called by each test with different parameters.